src/main/java/trello/api/rest/RestClientAPI.java

```diff
@@ -12,57 +12,106 @@
+import static io.restassured.http.ContentType.JSON;
 
 /**
- * RestClientAPI class.
+ * Authentication class.
  *
- * @author Josue Rodriguez.
- * @version 0.0.1
+ * @author Josue Rodriguez Garcia.
+ * @version 0.0.1.
  */
-public final class RestClientAPI {
+public class RestClientAPI {
+    private RequestSpecification requestSpecification;
+
+    /**
+     * Constructor method.
+     * @param requestSpecification defines a requestSpecification for input.
+     */
+    public RestClientAPI(final RequestSpecification requestSpecification) {
+        this.requestSpecification = requestSpecification;
+    }
+
+    /**
+     * Gets a response of type get.
+     * @param endpoint defines the type the http method.
+     * @return a response.
+     */
+    public Response get(final String endpoint) {
+        return apiResponse("GET", endpoint);
+    }
+
+    /**
+     * Gets a response of type put.
+     * @param endpoint defines the type the http method.
+     * @return a response.
+     */
+    public Response put(final String endpoint) {
+        return apiResponse("PUT", endpoint);
+    }
 
-    private static RestClientAPI oauth;
-    private RequestSpecification request;
+    /**
+     * Gets a response of type post.
+     * @param endpoint defines the type the http method.
+     * @return a response.
+     */
+    public Response post(final String endpoint) {
+        return apiResponse("POST", endpoint);
+    }
 
     /**
-     * This is constructor that initializes variables.
+     * Gets a response of type delete.
+     * @param endpoint defines the type the http method.
+     * @return a response.
      */
-    private RestClientAPI() {
-        User user = JsonConverter.jsonToUser(ReadJsonFile.getInstance().getDataByUserType("admin"));
-        String consumerKey = user.getConsumerKey();
-        String consumerSecret = user.getConsumerSecret();
-        String accessToken = user.getAccessToken();
-        String tokenSecret = user.getTokenSecret();
-        String baseUrl = NamePages.getBaseUrlAPI();
-        request = new RequestSpecBuilder().setAuth(RestAssured.oauth(consumerKey, consumerSecret,
-                accessToken, tokenSecret)).setBaseUri(baseUrl).build();
+    public Response delete(final String endpoint) {
+        return apiResponse("DELETE", endpoint);
     }
 
     /**
-     * Gives the class instance according Singleton pattern.
-     *
-     * @return an instance.
+     * Gets a requestSpecification for RestClientAPI.
+     * @return a requestSpecification.
      */
-    public static RestClientAPI getInstance() {
-        if (oauth == null) {
-            oauth = new RestClientAPI();
-        }
-        return oauth;
+    public RequestSpecification getRequest() {
+        return requestSpecification;
     }
 
     /**
-     * Gives the request specification resultant of oauth.
-     *
-     * @return an request specification.
+     * Sets the attribute requestSpecification en RestClientAPI.
```

Comment:
Please add a white space bellow this line.

Reply:
Solved